WordPressKit/WordPressKit/MediaServiceRemoteXMLRPC.m

```diff
@@ -30,13 +30,41 @@ - (void)getMediaWithID:(NSNumber *)mediaID
 - (void)getMediaLibraryWithSuccess:(void (^)(NSArray *))success
                            failure:(void (^)(NSError *))failure
 {
-    NSArray *parameters = [self defaultXMLRPCArguments];
+    NSMutableArray * mediaResult = [[NSMutableArray alloc] init];
+    [self getMediaLibraryStartOffset:0 media:mediaResult success:success failure:failure];
+}
+
+- (void)getMediaLibraryStartOffset:(NSUInteger)offset
+                             media:(NSArray *)media
+                           success:(void (^)(NSArray *))success
+                           failure:(void (^)(NSError *))failure
+{
+    NSInteger pageSize = 100;
+    NSDictionary *filter = @{
+                                @"number": @(pageSize),
+                                @"offset": @(offset)
+                            };
+    NSArray *parameters = [self XMLRPCArgumentsWithExtra:filter];
+
     [self.api callMethod:@"wp.getMediaLibrary"
               parameters:parameters
                  success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
                      NSAssert([responseObject isKindOfClass:[NSArray class]], @"Response should be an array.");
                      if (success) {
-                         success([self remoteMediaFromXMLRPCArray:responseObject]);
+                         NSArray *pageMedia = [self remoteMediaFromXMLRPCArray:responseObject];
+                         NSMutableArray *resultMedia = [NSMutableArray arrayWithArray:media];
+                         if (pageMedia.count > 0) {
+                             [resultMedia addObjectsFromArray: pageMedia];
+                         }
+                         // Did we got all the items we requested or it's finished?
+                         if (pageMedia.count < pageSize) {
+                             if (success) {
+                                 success(resultMedia);
+                             }
+                             return;
+                         }
+                         NSUInteger newOffset = offset + pageSize;
+                         [self getMediaLibraryStartOffset:newOffset media:resultMedia success: success failure: failure];
```

Comment:
Am I understanding this correctly and we are getting the entire Media Library for the site in one go? It looks like we are doing the same for .com sites, so performance should be tested already... I'm just surprised this works and doesn't kill the app for sites with a lot of media.

Reply:
@elibud so remember the media object it's basically an url with some extra metadata, it's doesn't include the bytes for media, so even a large media library (1000 objects) it's only going to be some kbs top. 
A single image download of a full size iPhoneX image will be a lot larger than all the size of the media library info.

I'm still surprised that the original request complained that the server went crashing when getting all the library, unless they are doing some extra processing on each media object before returning the result I don't see why it's happening.